packages/core/src/useNavigationBuilder.tsx

```diff
@@ -374,6 +369,12 @@ export default function useNavigationBuilder<
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  // We initialize this ref here to avoid a new getState getting initialized
+  // whenever initializedState changes. We want getState to have access to the
+  // latest initializedState, but don't need it to change when that happens
+  const initializedStateRef = React.useRef<State>();
+  initializedStateRef.current = initializedState;
```

Comment:
We're admittedly setting `initializedStateRef.current` in the body here. However, this shouldn't cause problems with Concurrent Mode because:

1. Setting `initializedStateRef` is an idempotent operation. In contrast with something like a ref being incremented, we aren't worried about it happening multiple times, so long as we have the correct value at the end
2. The hook itself only ever sets `initializedStateRef`. Since we never read from it, we don't have to worry about one concurrent render interfering with another

Also worth noting that this PR doesn't introduce this ref, it just moves its declaration down. In `master` we currently have the same ref being set during render, but it's done in a concurrent-unsafe way (relying on the current value for comparison)